### PR TITLE
feat: Add `include` option to `getObject` and `fetch`

### PR DESCRIPTION
**Pull request by mbfakourii**
### New Pull Request Checklist
- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

### Issue Description
Added ```include``` in ```getObject``` and ```fetch```
Added test

Related issue: #797

### Approach
In ```ParseObject``` class  and ```getObject``` and ```fetch``` functions there is no value ```include```

### TODOs before merging
- [x] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [ ] A changelog entry

### CODE REVIEW

**Comment by mtrezza** [comment]
(Sorry for tagging everyone for review here; but we didn't get any reviews on the last PR, so I hope someone could talk a look at this)

File: packages/dart/test/parse_object_test.dart [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    });
+
+    test('should return expectedIncludeResult json when use getObject',
+        () async {
+      // arrange
+      final MockParseClient client = MockParseClient();
+
+      await Parse().initialize(
+        'appId',
+        'https://test.parse.com',
```
Comment (mtrezza) [inline packages/dart/test/parse_object_test.dart:73]:
```suggestion
        'https://example.com',
```

File: packages/dart/test/parse_object_test.dart [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+@GenerateMocks([ParseClient])
+void main() {
+  group('parseObject', () {
+    test('should return expectedIncludeResult json when use fetch', () async {
+      // arrange
+      final MockParseClient client = MockParseClient();
+
+      await Parse().initialize(
+        'appId',
+        'https://test.parse.com',
```
[the remark this answers is not in the document]
Comment (mbfakourii) [reply packages/dart/test/parse_object_test.dart:16]: It is like this in other files. I change all the files in this way

File: packages/dart/test/parse_object_test.dart [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      myUserObject.objectId = "Mn1iJTkWTE";
+
+      when(client.get(
+        any,
+        options: anyNamed("options"),
+        onReceiveProgress: anyNamed("onReceiveProgress"),
+      )).thenAnswer((_) async => ParseNetworkResponse(
+          statusCode: 200,
+          data:
+              "{\"results\":[{\"objectId\":\"Mn1iJTkWTE\",\"phone\":\"+12025550463\",\"createdAt\":\"2022-09-04T13:35:20.883Z\",\"updatedAt\":\"2022-11-14T10:55:56.202Z\",\"img\":{\"objectId\":\"8nGrLj3Mvk\",\"size\":67663,\"mime\":\"image/jpg\",\"file\":{\"__type\":\"File\",\"name\":\"dc7320ee9146ee19aed8997722fd4e3c.bin\",\"url\":\"http://ip:port/api/files/myapp/dc7320ee9146ee19aed8997722fd4e3c.bin\"},\"createdAt\":\"2022-11-14T10:55:56.025Z\",\"updatedAt\":\"2022-11-14T10:55:56.025Z\",\"__type\":\"Object\",\"className\":\"MyFile\"}}]}"));
```
[the remark this answers is not in the document]
Comment (mbfakourii) [reply packages/dart/test/parse_object_test.dart:38]: It is true that it can be written better

[the remark this answers is not in the document]
Comment (mbfakourii) [reply packages/dart/test/parse_object_test.dart:38]: @mtrezza 
This json is a fake json from a fake server

[the remark this answers is not in the document]
Comment (mtrezza) [reply packages/dart/test/parse_object_test.dart:38]: Yes, but using raw JSON strings could make test debugging more difficult in the future or the test less robust. Could you create a normal object and stringily it to JSON?

File: packages/dart/test/parse_object_test.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (mtrezza) [reply packages/dart/test/parse_object_test.dart]: Got i. We shouldn't use such a domain, maybe you could open a PR to correct that with all other occurrences of that domain as well. The only "fake" domain we are using is `example.com`.

File: packages/dart/lib/src/network/parse_query.dart [hunk 4]
```diff
@@ -435,20 +435,6 @@ class QueryBuilder<T extends ParseObject> {
     return queryBuilder;
   }
 
-  String concatenateArray(List<String> queries) {
```
Comment (mtrezza) [inline packages/dart/lib/src/network/parse_query.dart:438]: Why is this removed?

File: packages/dart/test/parse_object_test.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbfakourii) [reply packages/dart/test/parse_object_test.dart:38]: No, because JSON is sent from the server and should be used to simulate server data. If we want to create an object, the test model must be completely changed.

File: packages/dart/lib/src/objects/parse_installation.dart [hunk 5]
```diff
@@ -85,6 +85,7 @@ class ParseInstallation extends ParseObject {
     set<String?>(keyLocaleIdentifier, ParseCoreData().locale);
 
     //Timezone
+    set<String>(keyTimeZone, _getNameLocalTimeZone());
```
Comment (mtrezza) [inline packages/dart/lib/src/objects/parse_installation.dart:88]: Is this change related to the feature "Add include option to getObject and fetch"?

File: packages/dart/lib/src/network/parse_query.dart (hunk 4 above)
Comment (mbfakourii) [reply packages/dart/lib/src/network/parse_query.dart:438]: transferred it to parse_utils.dart file for access in other files

File: packages/dart/lib/src/objects/parse_installation.dart (hunk 5 above)
Comment (mbfakourii) [reply packages/dart/lib/src/objects/parse_installation.dart:88]: No, it is for #616. The title should be changed

Comment (mtrezza) [reply packages/dart/lib/src/objects/parse_installation.dart:88]: A PR should only address one issue, unless they are directly related as a consequence of one another. Could you split this into 2 PRs?

File: packages/dart/lib/src/network/parse_query.dart (hunk 4 above)
Comment (mtrezza) [reply packages/dart/lib/src/network/parse_query.dart:438]: Got it, I've overlooked that

File: packages/dart/lib/src/objects/parse_installation.dart (hunk 5 above)
Comment (mbfakourii) [reply packages/dart/lib/src/objects/parse_installation.dart:88]: What title do you suggest? I have no idea

File: packages/dart/test/parse_object_test.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (mtrezza) [reply packages/dart/test/parse_object_test.dart:38]: I don't mean a Parse object, I mean a plain object and for example:

```dart
import 'dart:convert';
var response = {
     "key":"value"
};
String fakeServerResponseJson = jsonEncode(fakeServerResponse);
```

File: packages/dart/lib/src/objects/parse_installation.dart (hunk 5 above)
Comment (mbfakourii) [reply packages/dart/lib/src/objects/parse_installation.dart:88]: > A PR should only address one issue, unless they are directly related as a consequence of one another. Could you split this into 2 PRs?

Sorry, I didn't know about this topic. I separate the PRs

Comment (mtrezza) [reply packages/dart/lib/src/objects/parse_installation.dart:88]: No worries, thanks!

File: packages/dart/test/parse_object_test.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (mtrezza) [reply packages/dart/test/parse_object_test.dart:38]: Thanks; the idea is that it's easier to read, modify and debug in the future. We usually never use raw JSON strings in tests but convert them from an object.

File: packages/dart/lib/src/objects/parse_installation.dart (hunk 5 above)
Comment (mbfakourii) [reply packages/dart/lib/src/objects/parse_installation.dart]: i reverted to back

Comment (mtrezza) [reply packages/dart/lib/src/objects/parse_installation.dart]: thanks i'll resolve this

File: packages/dart/test/parse_object_test.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbfakourii) [reply packages/dart/test/parse_object_test.dart:38]: This method works, I will try to improve all the tests

[the remark this answers is not in the document]
Comment (mbfakourii) [reply packages/dart/test/parse_object_test.dart]: All tests improved

[the remark this answers is not in the document]
Comment (mtrezza) [reply packages/dart/test/parse_object_test.dart]: Looks good!

**Comment by mbfakourii** [comment]
> Could you bump the version and add the cangelog?

added

File: packages/dart/CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,9 @@
+## [3.1.7](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-3.1.6...dart-3.1.7) (2022-12-22)
+
+### Bug Fixes
+
+* feat: Add `include` option to `getObject` and `fetch` ([#798](https://github.com/parse-community/Parse-SDK-Flutter/issues/798))
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:5]:
```suggestion
### Features

* Add `include` option to `getObject` and `fetch` ([#798](https://github.com/parse-community/Parse-SDK-Flutter/issues/798))
```

**Review by mtrezza — CHANGES_REQUESTED** [review]
